specs/release.md

```diff
@@ -0,0 +1,319 @@
+3. **Wrangler CLI:** `wrangler rollback` command for local rollback operations
+
+Manual rollback SHOULD be used for issues discovered after deployment verification passes (e.g., subtle bugs, performance degradation, user reports).
+
+Rollback MUST NOT require rebuilding, retesting, or code changes. Cloudflare Workers' deployment history provides instant rollback to previous deployments.
+
+## Pre-Deployment Requirements
+
+Before any deployment can succeed, certain conditions MUST be satisfied. The CI/CD pipeline enforces these requirements automatically.
+
+### Code Quality Gates
+
+All code merged to `main` MUST meet quality standards:
+
+- Zero TypeScript type errors
+- Zero linter errors (when linter is configured)
+- 95%+ statement coverage
+- 95%+ function coverage
+- All tests passing (unit and E2E)
+- No high-severity security vulnerabilities in dependencies
+
+Pull requests MUST NOT be merged if any quality gate fails.
+
+### Dependency Management
+
+The project uses `pnpm` with a lockfile (`pnpm-lock.yaml`) to ensure reproducible installations. Deployments MUST use `pnpm install --frozen-lockfile` to prevent unexpected dependency version changes.
+
+Dependency updates MUST be tested in development before production deployment. Automated dependency updates (e.g., Dependabot) SHOULD be configured to create pull requests for review, not auto-merge.
+
+### Configuration Validation
+
+Before deployment, the workflow SHOULD validate:
+
+- Required secrets are configured (GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET, etc.)
+- wrangler.toml contains valid configuration for target environment
+- Required Cloudflare resources exist (R2 buckets, KV namespaces, Durable Objects)
+
+Configuration errors SHOULD fail deployment before code is uploaded to prevent partial or broken deployments.
+
+## Deployment Notifications
+
+The CI/CD system MUST provide visibility into deployment status through multiple channels.
+
+### GitHub Actions Summary
+
+Every deployment MUST create a GitHub Actions step summary with:
+
+- Environment deployed to
+- Commit SHA deployed
+- Deployment timestamp
+- Deployment URL
+- Health check status
+- Link to Cloudflare Workers dashboard
+
+### Pull Request Comments
+
+When a pull request is merged and triggers development deployment, the workflow SHOULD post a comment to the pull request with deployment status and URL for testing.
+
+### Deployment History
+
+GitHub Deployments API provides centralized deployment history viewable at:
+
+`https://github.com/{owner}/{repo}/deployments`
+
+This history MUST be maintained for all deployments and MUST include:
+
+- Commit SHA deployed
+- Environment name
+- Deployment status (success/failure/in_progress)
+- Deployment URL
+- Timestamp
+
+## Release Documentation
+
+While deployments do not require release notes, significant production releases SHOULD be documented.
+
+### Changelog Maintenance
```

Comment:
I love the idea of this but I have no idea how this will work in practice. I'm not manually doing it so unless we can build it into the process it is not going to happen.  The best way I've seen this done is each feature branch had a way of recoded what it did in a file somewhere and then you have a particular release process that combines these into the changelog entry for that release and decides things like version number increment, etc. honestly for what we're building I think it's overkill. Got can be our changelog for now and if we ever want better than we need a proper release stage with a release branch where we can do some of this stuff. Unless you have a better idea.